From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 8 Oct 2026 21:10:46 -0700
Subject: [PATCH] fixup! Moonrise optimisation patches

Moonrise: Avoid oversized block counting lists for global palettes (7a16dbe)

diff --git a/net/minecraft/util/SimpleBitStorage.java b/net/minecraft/util/SimpleBitStorage.java
index b775c716653b476da5aa039a1810e741a503509a..bc48900cc4aa557852a9279649aac4e336f32007 100644
--- a/net/minecraft/util/SimpleBitStorage.java
+++ b/net/minecraft/util/SimpleBitStorage.java
@@ -460,7 +460,7 @@ public class SimpleBitStorage implements BitStorage {
 
             return ret;
         } else {
-            final java.util.List<ca.spottedleaf.moonrise.patches.block_counting.BlockCountingEntry> ret = new java.util.ArrayList<>(1 << bits);
+            final java.util.List<ca.spottedleaf.moonrise.patches.block_counting.BlockCountingEntry> ret = new java.util.ArrayList<>(1 << 6);
             final it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap<ca.spottedleaf.moonrise.patches.block_counting.BlockCountingEntry> byPaletteId = new it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap<>(
                 1 << 6
             );
